src/amo/components/InstallWarning/index.js

```diff
@@ -0,0 +1,184 @@
+/* @flow */
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import type { AppState } from 'amo/store';
+import { ADDON_TYPE_EXTENSION, UNINSTALLED, UNKNOWN } from 'core/constants';
+import translate from 'core/i18n/translate';
+import log from 'core/logger';
+import tracking from 'core/tracking';
+import { withExperiment } from 'core/withExperiment';
+import Notice from 'ui/components/Notice';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+import type { InstalledAddon } from 'core/reducers/installations';
+import type { WithExperimentInjectedProps } from 'core/withExperiment';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType,
+|};
+
+type CouldShowWarningParams = {|
+  addonIsExtension: boolean,
+  addonIsRecommended: boolean | void,
+  experimentEnabled: $PropertyType<
+    WithExperimentInjectedProps,
+    'experimentEnabled',
+  >,
+  installStatus: $PropertyType<InstalledAddon, 'status'>,
+|};
+
+export const couldShowWarning = ({
+  addonIsExtension,
+  addonIsRecommended,
+  experimentEnabled,
+  installStatus,
+}: CouldShowWarningParams) => {
+  return (
+    addonIsExtension &&
+    !addonIsRecommended &&
+    experimentEnabled &&
+    [UNINSTALLED, UNKNOWN].includes(installStatus)
+  );
+};
+
+const addonIsExtension = (addon: AddonType) => {
+  return addon.type === ADDON_TYPE_EXTENSION;
+};
+
+type InternalProps = {|
+  ...Props,
+  ...WithExperimentInjectedProps,
+  _couldShowWarning: typeof couldShowWarning,
+  _log: typeof log,
+  _tracking: typeof tracking,
+  i18n: I18nType,
+  installStatus: $PropertyType<InstalledAddon, 'status'>,
+|};
+
+export const EXPERIMENT_CATEGORY_CLICK =
+  'AMO Install Button Warning Experiment - Click';
+export const EXPERIMENT_CATEGORY_SHOW =
+  'AMO Install Button Warning Experiment - Display';
+export const EXPERIMENT_ID = 'installButtonWarning';
+export const INSTALL_WARNING_EXPERIMENT_DIMENSION = 'dimension6';
+export const VARIANT_INCLUDE_WARNING = 'includeWarning';
+export const VARIANT_EXCLUDE_WARNING = 'excludeWarning';
+
+export class InstallWarningBase extends React.Component<InternalProps> {
+  static defaultProps = {
+    _couldShowWarning: couldShowWarning,
+    _log: log,
+    _tracking: tracking,
+  };
+
+  maybeSendDisplayTrackingEvent = () => {
+    const {
+      _couldShowWarning,
+      _tracking,
+      addon,
+      experimentEnabled,
+      installStatus,
+      variant,
+    } = this.props;
+
+    if (
+      _couldShowWarning({
+        addonIsExtension: addonIsExtension(addon),
```

Comment:
I feel like the `addonIsExtension` helper is superfluous.

Reply:
I do call it twice, so we'd have the logic twice, but it's true that it's more lines of code to have the helper than not have it, and it's not unusual logic, so I'll remove it.